main.go

```diff
@@ -48,115 +45,42 @@ func main() {
-	ev := libhoney.NewEvent()
-
-	// calculate event duration
-	// TODO: This is a hack to work around a bug that results in the response timestamp sometimes
-	// being zero which causes the event duration to be negative.
-	if event.RequestTimestamp.IsZero() {
-		log.Debug().
-			Str("stream_ident", event.StreamIdent).
-			Int64("request_id", event.RequestId).
-			Msg("Request timestamp is zero")
-		ev.AddField("http.request.timestamp_missing", true)
-		event.RequestTimestamp = time.Now()
-	}
-	if event.ResponseTimestamp.IsZero() {
-		log.Debug().
-			Str("stream_ident", event.StreamIdent).
-			Int64("request_id", event.RequestId).
-			Msg("Response timestamp is zero")
-		ev.AddField("http.response.timestamp_missing", true)
-		event.ResponseTimestamp = time.Now()
-	}
-	eventDuration := event.ResponseTimestamp.Sub(event.RequestTimestamp)
-
-	// common attributes
-	ev.Timestamp = event.RequestTimestamp
-	ev.AddField("meta.httpEvent_handled_at", time.Now())
-	ev.AddField("meta.httpEvent_request_handled_latency_ms", time.Since(event.RequestTimestamp).Milliseconds())
-	ev.AddField("meta.httpEvent_response_handled_latency_ms", time.Since(event.ResponseTimestamp).Milliseconds())
-	ev.AddField("meta.stream.ident", event.StreamIdent)
-	ev.AddField("duration_ms", eventDuration.Milliseconds())
-	ev.AddField("http.request.timestamp", event.RequestTimestamp)
-	ev.AddField("http.response.timestamp", event.ResponseTimestamp)
-
-	ev.AddField(string(semconv.ClientSocketAddressKey), event.SrcIp)
-	ev.AddField(string(semconv.ServerSocketAddressKey), event.DstIp)
-
-	var requestURI string
-
-	// request attributes
-	if event.Request != nil {
-		requestURI = event.Request.RequestURI
-		ev.AddField("name", fmt.Sprintf("HTTP %s", event.Request.Method))
-		ev.AddField(string(semconv.HTTPRequestMethodKey), event.Request.Method)
-		ev.AddField(string(semconv.URLPathKey), requestURI)
-		ev.AddField(string(semconv.UserAgentOriginalKey), event.Request.Header.Get("User-Agent"))
-		ev.AddField(string(semconv.HTTPRequestBodySizeKey), event.Request.ContentLength)
-	} else {
-		ev.AddField("name", "HTTP")
-		ev.AddField("http.request.missing", "no request on this event")
-	}
-
-	// response attributes
-	if event.Response != nil {
-		ev.AddField(string(semconv.HTTPResponseStatusCodeKey), event.Response.StatusCode)
-		ev.AddField(string(semconv.HTTPResponseBodySizeKey), event.Response.ContentLength)
-
-	} else {
-		ev.AddField("http.response.missing", "no response on this event")
-	}
+	shutdownChannel := make(chan bool, 1)
 
-	ev.Add(utils.GetK8sAttrsForIp(k8sClient, event.SrcIp, "source"))
-	ev.Add(utils.GetK8sAttrsForIp(k8sClient, event.DstIp, "destination"))
+	// cleanup func that waits for for shutdown signal and stops the assembler, event handler and k8s client
+	go func() {
+		<-signalChannel
+		log.Info().Msg("Agent is stopping. Cleaning up...")
+		// signal the k8sClient, event handler and assembler to stop via ctx.Done()
+		done()
+		<-shutdownChannel
```

Comment:
```suggestion
		shutdownChannel <- true
```

Publish here, right?